File: CexIOPythonV1/cexio.py
import requests
from colorama import init, Fore, Style
import time
import random
from datetime import datetime
init(autoreset=True)

start_text = """
█▀▀ █▀▀ ▀▄▀ █ █▀█
█▄▄ ██▄ █░█ █ █▄█
"""
# Reusable headers
HEADERS = {
    "origin": "https://cexp.cex.io",
    "referer": "https://cexp.cex.io"
}

BASE_URL = "https://cexp.cex.io/api/"
DEV_AUTH_DATA = 906306292

def make_post_request(endpoint, payload):
    url = BASE_URL + endpoint
    try:
        response = requests.post(url, headers=HEADERS, json=payload)
        response.raise_for_status()  # Raise HTTPError for non-200 status codes
        return response.json()
    except requests.exceptions.RequestException as e:
        #print(f"{Fore.RED + Style.BRIGHT}Error during {endpoint}: {e}")
        return None

def get_user_info(query_id):
    payload = {
        "devAuthData": DEV_AUTH_DATA,
        "authData": query_id,
        "data": {},
        "platform": "android"
    }
    return make_post_request("getUserInfo", payload)

def claim_farm(query_id):
    payload = {
        "devAuthData": DEV_AUTH_DATA,
        "authData": query_id,
        "data": {},
        "platform": "android"
    }
    return make_post_request("claimFarm", payload)

def start_farm(query_id):
    payload = {
        "devAuthData": DEV_AUTH_DATA,
        "authData": query_id,
        "data": {},
        "platform": "android"
    }
    return make_post_request("startFarm", payload)

def claim_taps(query_id, taps):
    payload = {
        "authData": query_id,
        'data': {'taps': taps},
    }
    return make_post_request("claimTaps", payload)

def countdown(secs):
    for i in range(secs, 0, -1):
        print(f"\r{Fore.MAGENTA+Style.BRIGHT}Sleeping for {i} seconds...", end="", flush=True)
        time.sleep(1)
    print("\r" + " " * 50, end="", flush=True)  # Clear the countdown message
    print("\n")  # Print a newline to ensure the prompt appears on a new line

def read_query_id(filename):
    with open(filename, "r") as file:
        return file.read().splitlines()

def main():
    filename = "query_id.txt"
    query_ids = read_query_id(filename)

    while True:
        #print(start_text)
        for query_id in query_ids:
            print(f"{Fore.YELLOW + Style.BRIGHT}[CEXP] [{datetime.now().strftime('%H:%M:%S')}] {query_id[:20] + '...' if len(query_id) > 20 else query_id}...")
            
            user_info = get_user_info(query_id)
            if user_info:
                balance = user_info['data'].get('balance', 'N/A')
                first_name = user_info['data'].get('first_name', 'N/A')
                last_name = user_info['data'].get('last_name', 'N/A')
                print(f"{Fore.GREEN + Style.BRIGHT}Name: {first_name} {last_name}  Balance: {balance} CEXP")
            else:
                print(f"{Fore.RED + Style.BRIGHT}User information is not available.")

            farm_info = claim_farm(query_id)
            if farm_info:
                print(f"{Fore.GREEN + Style.BRIGHT}Status: {farm_info['status']}")
                print(f"{Fore.GREEN + Style.BRIGHT}Balance: {farm_info['data'].get('balance', 'N/A')}")
                print(f"{Fore.GREEN + Style.BRIGHT}Claimed Balance: {farm_info['data'].get('claimedBalance', 'N/A')}")
            else:
                print(f"{Fore.BLUE + Style.BRIGHT}->Claim Farm: Not available.")
                
            start_info = start_farm(query_id)
            if start_info:
                print(f"{Fore.GREEN + Style.BRIGHT}->Start Farm: {start_info['status']}")
            else:
                print(f"{Fore.BLUE + Style.BRIGHT}->Start Farm: Not available.")
                
            claim_info = claim_taps(query_id, 5)
            if claim_info:
                print(f"{Fore.GREEN + Style.BRIGHT}->Claim Taps: {claim_info['status']}")
                available_taps = claim_info['data'].get('availableTaps', 0)
                print(f"{Fore.GREEN + Style.BRIGHT}->Remain Taps: {available_taps}")
            else:
                print(f"{Fore.BLUE + Style.BRIGHT}->Claim Taps: Not available.")
        
        random_delay = random.randint(300, 500)
        countdown(random_delay)

if __name__ == "__main__":
    main()

File: CexIONode/cexio.js
const axios = require('axios');
const fs = require('fs');
const colors = require('colors');

const BASE_URL = "https://cexp.cex.io/api/";
const DEV_AUTH_DATA = 906306292;

const headers = {
    "origin": "https://cexp.cex.io",
    "referer": "https://cexp.cex.io"
};

async function makePostRequest(endpoint, payload) {
    const url = BASE_URL + endpoint;
    try {
        const response = await axios.post(url, payload, { headers });
        return response.data;
    } catch (error) {
        // console.error(colors.red.bold(`Error during ${endpoint}: ${error}`));
        return null;
    }
}

async function getUserInfo(query_id) {
    const payload = {
        "devAuthData": DEV_AUTH_DATA,
        "authData": query_id,
        "data": {},
        "platform": "android"
    };
    return await makePostRequest("getUserInfo", payload);
}

async function claimFarm(query_id) {
    const payload = {
        "devAuthData": DEV_AUTH_DATA,
        "authData": query_id,
        "data": {},
        "platform": "android"
    };
    return await makePostRequest("claimFarm", payload);
}

async function startFarm(query_id) {
    const payload = {
        "devAuthData": DEV_AUTH_DATA,
        "authData": query_id,
        "data": {},
        "platform": "android"
    };
    return await makePostRequest("startFarm", payload);
}

async function claimTaps(query_id, taps) {
    const payload = {
        "authData": query_id,
        'data': {'taps': taps}
    };
    return await makePostRequest("claimTaps", payload);
}

function countdown(secs) {
    let interval = setInterval(() => {
        if (secs > 0) {
            process.stdout.write(`\r${colors.magenta.bold(`Sleeping for ${secs} seconds...`)}`);
            secs--;
        } else {
            clearInterval(interval);
            process.stdout.write("\r" + " ".repeat(50));
            console.log("\n");
        }
    }, 1000);
}

function readQueryIds(filename) {
    try {
        const data = fs.readFileSync(filename, 'utf8');
        return data.split('\n').map(line => line.trim()).filter(line => line && !line.startsWith('#'));
    } catch (error) {
        console.error(colors.red.bold(`Error reading file ${filename}`));
        return [];
    }
}

function parseQueryString(queryString) {
    return queryString.split('&').reduce((acc, part) => {
        const [key, value] = part.split('=');
        acc[decodeURIComponent(key)] = decodeURIComponent(value);
        return acc;
    }, {});
}

async function main() {
    const filename = "query_id.txt";
    const queryIds = readQueryIds(filename);
    while (true) {
        for (const query_id of queryIds) {
            console.log(colors.yellow.bold(`[CEXP] [${new Date().toLocaleTimeString()}] ${query_id.length > 20 ? query_id.slice(0, 20) + '...' : query_id}...`));

            const userInfo = await getUserInfo(query_id);
            if (userInfo) {
                const { balance = 'N/A', first_name = 'N/A', last_name = 'N/A' } = userInfo.data;
                console.log(colors.green.bold(`Name: ${first_name} ${last_name}  Balance: ${balance} CEXP`));
            } else {
                console.log(colors.red.bold("User information is not available."));
            }

            const farmInfo = await claimFarm(query_id);
            if (farmInfo) {
                console.log(colors.green.bold(`Status: ${farmInfo.status}`));
                console.log(colors.green.bold(`Balance: ${farmInfo.data?.balance || 'N/A'}`));
                console.log(colors.green.bold(`Claimed Balance: ${farmInfo.data?.claimedBalance || 'N/A'}`));
            } else {
                console.log(colors.blue.bold("->Claim Farm: Not available."));
            }

            const startInfo = await startFarm(query_id);
            if (startInfo) {
                console.log(colors.green.bold(`->Start Farm: ${startInfo.status}`));
            } else {
                console.log(colors.blue.bold("->Start Farm: Not available."));
            }

            const claimInfo = await claimTaps(query_id, 5);
            if (claimInfo) {
                console.log(colors.green.bold(`->Claim Taps: ${claimInfo.status}`));
                const availableTaps = claimInfo.data?.availableTaps || 0;
                console.log(colors.green.bold(`->Remain Taps: ${availableTaps}`));
            } else {
                console.log(colors.blue.bold("->Claim Taps: Not available."));
            }
        }

        const randomDelay = Math.floor(Math.random() * (500 - 300 + 1)) + 300;
        countdown(randomDelay);
        await new Promise(resolve => setTimeout(resolve, randomDelay * 1000));
    }
}

main();
